packages/game-passes/contracts/GamePasses.sol

```diff
@@ -0,0 +1,1161 @@
+        _grantRole(OPERATOR_ROLE, _operator);
+        _grantRole(SIGNER_ROLE, _signer);
+        paymentToken = _paymentToken;
+        baseURI = _baseURI;
+        defaultTreasuryWallet = _defaultTreasury;
+        _setDefaultRoyalty(_royaltyReceiver, _royaltyFeeNumerator);
+        DOMAIN_SEPARATOR = keccak256(
+            abi.encode(
+                EIP712_DOMAIN_TYPEHASH,
+                keccak256(bytes("SandboxPasses1155")),
+                keccak256(bytes("1")),
+                block.chainid,
+                address(this)
+            )
+        );
+    }
+
+    // =============================================================
+    //                      External & Public Functions
+    // =============================================================
+
+    /**
+     * @notice Mints tokens with a valid EIP-712 signature, requiring payment in the configured token
+     * @param from Address that will receive the tokens (must be same as msg.sender)
+     * @param tokenId ID of the token to mint
+     * @param amount Number of tokens to mint
+     * @param price Price to pay in payment token units
+     * @param deadline Timestamp after which the signature becomes invalid
+     * @param signature EIP-712 signature from an authorized signer
+     * @dev Verifies the signature, checks supply limits, processes payment, and mints tokens
+     * @dev Updates the per-wallet minting count and transfers payment to the appropriate treasury
+     * @dev Reverts if:
+     *      - Contract is paused
+     *      - Token is not configured
+     *      - Signature is invalid or expired
+     *      - Max supply would be exceeded
+     *      - Max per wallet would be exceeded
+     *      - Payment transfer fails
+     */
+    function mint(
+        address from,
+        uint256 tokenId,
+        uint256 amount,
+        uint256 price,
+        uint256 deadline,
+        bytes memory signature
+    ) external whenNotPaused {
+        address caller = _msgSender();
+
+        if (from != caller) {
+            revert InvalidSender(from, caller);
+        }
+
+        TokenConfig storage config = tokenConfigs[tokenId];
+
+        if (!config.isConfigured) {
+            revert TokenNotConfigured(tokenId);
+        }
+        MintRequest memory request = MintRequest({
+            caller: caller,
+            tokenId: tokenId,
+            amount: amount,
+            price: price,
+            deadline: deadline,
+            nonce: nonces[caller]++
+        });
+
+        verifySignature(request, signature);
+        _checkMaxPerWallet(tokenId, caller, amount);
+        _checkMaxSupply(tokenId, amount);
+
+        // Update minted amount for wallet
+        config.mintedPerWallet[caller] += amount;
+
+        // Transfer payment
+        address treasury = config.treasuryWallet != address(0) ? config.treasuryWallet : defaultTreasuryWallet;
+        SafeERC20.safeTransferFrom(IERC20(paymentToken), _msgSender(), treasury, price);
```

Comment:
maybe it is possible to use `_processSingleMint`  here ?

Reply:
Yes! Absolutely, i think I extracted the logic and forgot to update the function body.
Implemented: https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/ac6375dbdfbd9060452059af0e835bfe926c25c7